### DISCUSSION TITLE
Azure DevOps vs Github Enterprise for Copilot

### DISCUSSION THREAD

**Discussion by zaffaste** [community/community#88693] [Enterprise]
Hello everyone and thanks in advance for your help,
we currently use Azure DevOps with git code repository fully integrated into Visual Studio and Visual Studio Code. Now we want to add the new Github Copilot functionality and as far as I know we have to pay for a Github enterprise subscription (whose features, as far as I know, are completely covered by Azure DevOps) and then we have to join the waitlist before paying to use Github co-pilot. Are these considerations correct?
If my considerations above are correct, what might be the features we would have with Github Enterprise that we didn't actually have with Azure DevOps?

Thank you so much,
Stefano

**Comment by DarrenChochian** [comment]


**Reply by DonShrout** [reply]
@DarrenChochian Our company also has our entire codebase in Azure DevOps git repositorys. If we want to use GitHub Copilot Enterprise, we will have to purchase GitHub Enterprise Cloud subscriptions. Will that give Copilot access to our git repositories in Azure DevOps?

**Reply by DuckScapePhilip** [reply]
This is also a concern of mine. Is there a way to pay for GitHub CoPilot Enterprise and have it access our code base and wiki inside of Azure DevOps?

**Reply by react-o-licious** [reply]
> This is also a concern of mine. Is there a way to pay for GitHub CoPilot Enterprise and have it access our code base and wiki inside of Azure DevOps?

Still very interested in training copilot on our Azure DevOps repositories.

**Reply by Niladri24dutta** [reply]
@DarrenChochian is this supported yet by Azure Devops?

**Reply by Tiberriver256** [reply]
Would love to see GitHub CoPilot integrated into the licensing for Azure DevOps. GitHub Enterprise is a little too expensive to pay for if all it gets us is CoPilot in our IDE, at least relative to the cost experienced by those already using GitHub Enterprise. For them, it was  like a free feature. For those of us on Azure DevOps it's like doubling our costs.

**Reply by VBaudot** [reply]
+1

**Comment by wesfincher** [comment]
Why the silence here? How do we get Copilot in Azure DevOps without potentially doubling our spend? The documentation seems confusing. What's the minimal number of licenses needed to use Copilot in our IDE when on Azure DevOps?
